conda/core/index.py

```diff
@@ -158,24 +221,39 @@ def _make_virtual_package(name, version=None, build_string=None):
     )
 
 
-def _supplement_index_with_features(index, features=()):
+def _supplement_index_with_features(
+    index: dict[Any, Any], features: list[str] = []
+) -> None:
+    """
+    Supplement the given index with virtual feature records.
+
+    :param index: The package index to supplement.
+    :param features: A list of feature names to add to the index.
+    """
     for feature in chain(context.track_features, features):
         rec = make_feature_record(feature)
         index[rec] = rec
 
 
-def _supplement_index_with_system(index):
+def _supplement_index_with_system(index: dict[Any, Any]) -> None:
```

Comment:
```suggestion
def _supplement_index_with_system(index: dict[PackageRecord, PackageRecord]) -> None:
```

Same thing as above.